autorest/adal/token_1.13.go

```diff
@@ -34,3 +35,43 @@ func getMSIEndpoint(ctx context.Context, sender Sender) (*http.Response, error)
 	req.URL.RawQuery = q.Encode()
 	return sender.Do(req)
 }
+
+// EnsureFreshWithContext will refresh the token if it will expire within the refresh window (as set by
```

Comment:
Just for my own knowledge..why are these methods repeated in both token_1.13.go and token_legacy.go?

Reply:
If you look at the top of the file you'll notice a build tag.

`// +build go1.13`

This file will only compile for Go versions 1.13 and above.

In the legacy file, the build tag is `// +build !go1.13` meaning it will only compile for Go versions before 1.13.

With this combination we provide different implementations for all versions of Go.

Reply:
Note that in the 1.13 impl we use the `%w` format specifier which was added in Go 1.13.

Reply:
Ah got it, that's good to know. Thanks for the explanation :)